OKE LA2 Updates (#408)
canonical/ubuntu-cloud-docs: docs/reuse/OKE-nodes.txt

==============================================================

Start: Get Ubuntu image access

Get Ubuntu image access
---------------------------

Ubuntu worker node images for OKE are currently in a Limited Availability (LA) release. To gain access to the images, reach out to your Oracle account or support teams to express interest in this LA. 

Once access has been provided to your tenancy and region then you must add an additional `Policy`_ to your root compartment to permit the listing of these images. Below are the required contents of the `Policy`_.

.. code:: console

    Define tenancy oke as ocid1.tenancy.oc1..aaaaaaaa5vrtu4bjcqpjvbworiwffgccrgrbkum64mtn33yrccjrqpzuyara 
    Endorse any-user to read instance-images in tenancy oke

.. note:: 
    
    The tenancy OCID listed above for the `Policy`_ belongs to the OCI OKE Service, it is not owned by Canonical.

End: Get Ubuntu image access

Start: Find an image

Find an Ubuntu image
-----------------------

Once your tenancy has been whitelisted, you will be able to list the images from API or view them through the OCI Console. 

.. tabs::

    .. group-tab:: API

>         Prior to running any ``oci`` commands, be sure that your ``~/.oci/config`` is configured to use your whitelisted region.

Reviewer: s/whitelisted/allowlisted/

==============================================================

Start: Get Ubuntu image access

Get Ubuntu image access
---------------------------

.. attention::
    Ubuntu worker node images for OKE are currently in a Limited Availability (LA) release. To gain access to the images, reach out to your Oracle account or support teams to express interest in this LA. 

Once access has been provided to your tenancy and region then you must add an additional `Policy`_ to your root compartment to permit the listing of these images. Below are the required contents of the `Policy`_.

.. code:: console

    Define tenancy oke as ocid1.tenancy.oc1..aaaaaaaa5vrtu4bjcqpjvbworiwffgccrgrbkum64mtn33yrccjrqpzuyara 
    Endorse any-user to read instance-images in tenancy oke

.. note:: 
    
    The tenancy OCID listed above for the `Policy`_ belongs to the OCI OKE Service, it is not owned by Canonical.

End: Get Ubuntu image access

Start: Find an image

Find an Ubuntu image
-----------------------

> Once your tenancy has been enabled for the LA, you will be able to list the images from API or view them through the OCI Console. 

Author: Technically the CLI uses the API so when referencing "API" I meant that transitively they could be listed elsewhere.

This is confusing though, so I'll be more explicit.

.. tabs::

    .. group-tab:: API

        Prior to running any ``oci`` commands, be sure that your ``~/.oci/config`` is configured to use your region specified for the LA enablement.

>         Otherwise, start by listing all OKE compatible images and savings them to the ``node-pool-options.json`` file. The file will contain a list of all available OKE images including your newly added Ubuntu images.

Reviewer 2: ```suggestion
        Otherwise, start by listing all OKE compatible images and save them to the ``node-pool-options.json`` file. The file will contain a list of all available OKE images including your newly added Ubuntu images.
```

    
        .. code:: bash
            
            oci ce node-pool-options get --compartment-id <compartment-id> --node-pool-option-id all | tee node-pool-options.json
        

>         Use the following to filter it down to only the newly added Ubuntu images.

Reviewer 2: ```suggestion
        To filter it down to the newly added Ubuntu images, use:
```

        
        .. code:: bash
        
            jq -r '.data.sources[] | select(."source-name" | contains("ubuntu"))' node-pool-options.json
        

>         The results should be comprised of a list of objects that contain an ``image-id``, ``source-name`` and ``source-type``.

Reviewer 2: ```suggestion
        The result will contain objects of the form:
```

        
        .. code:: json
            
            {
              "image-id": "ocid1.image.oc1.phx.aaaaaaaajqzb5jcbcvh5obyg2l2hzzw5qewwhrknvlyb3zaglduivigvo4sq",
              "source-name": "ubuntu-amd64-minimal-22.04-jammy-v20250604.1-OKE-1.31.1",
              "source-type": "IMAGE"
            }

        Make note of the ``image-id`` for the image(s) that you wish to use since they will be required in later steps.

    .. group-tab:: Console
        

>         Whether you decide to create nodes using :guilabel:`Create Cluster` > :guilabel:`Quick Create` or :guilabel:`Add node pool`, both will let you select your desired node image. That list will contain all available OKE images and including your newly added Ubuntu images. 

Reviewer 2: ```suggestion
        Whether you decide to create nodes using :guilabel:`Create Cluster` > :guilabel:`Quick Create` or :guilabel:`Add node pool`, both will let you select your desired node image. That list will contain all available OKE images, including your newly added Ubuntu images. 
```

        .. image:: deploy-ubuntu-oke-nodes/oke-browse.png
           :align: center
        
        After selecting your desired image based on the architecture, Ubuntu suite and Kubernetes version, your configuration should look similar to the image below.

        .. image:: deploy-ubuntu-oke-nodes/oke-create-cluster.png
           :align: center

        .. warning::
            
            Pay attention to which image you select and ensure it's architecture corresponds to your selected pod shape.

>         When launching Ubuntu nodes from the Console during the LA, it's required to provide custom cloud-init ``user-data`` to the instances to ensure they join the cluster properly.

Reviewer 2: ```suggestion
        When launching Ubuntu nodes from the console during the LA, it's required to provide custom cloud-init ``user-data`` to the instances to ensure they join the cluster properly.
```

        Managed node ``user-data`` is quite simple:
        
        .. code:: yaml
           
            #cloud-config
        
            runcmd:
            - oke bootstrap
        

>         While Self-Managed ``user-data`` requires variable substitution for the cluster certificate and private control plane IP:

Reviewer 2: ```suggestion
        For Self-Managed ``user-data``, use the following with appropriate variable substitutions for the cluster certificate (``cluster_ca_cert``) and private control plane IP (``api_server_endpoint``):
```

        
        .. code:: yaml
        
           #cloud-config
        
            runcmd:
              - oke bootstrap --ca ${cluster_ca_cert} --apiserver-host ${api_server_endpoint}
            
            write_files:
              - path: /etc/oke/oke-apiserver
                permissions: '0644'
                content: ${api_server_endpoint}
              - encoding: b64
                path: /etc/kubernetes/ca.crt
                permissions: '0644'
                content: ${cluster_ca_cert}

>         Copy the ``user-data`` required for your node type and paste it into the Console.

Reviewer 2: ```suggestion
        Copy the ``user-data`` required for your node type and paste it into the console.
```
